src/main/java/programminglife/model/Graph.java

```diff
@@ -0,0 +1,127 @@
+package programminglife.model;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.*;
+
+/**
+ * Created by marti_000 on 25-4-2017.
+ */
+public class Graph {
+    private String id;
+    private Set<Node> rootNodes;
+    private List<Node> nodes;
+
+    public Graph(String id) {
+        this.nodes = new ArrayList<>();
+        this.nodes.add(new Node(-1));
+        this.id = id;
+        this.rootNodes = new HashSet<>();
+    }
+
+    public boolean addNode(Node node) {
+        return this.nodes.add(node);
```

Comment:
This should be `set(node.getId(), node)`.
It also needs a check that that position is not yet used (and that the array is big enough, with ensureCapacity?)